src/preact/context.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createContext, createElement} from 'preact';
+import {useContext} from 'preact/hooks';
+
+/**
+ * @typedef {{
+ *   Provider: *,
+ *   Consumer: *,
+ * }}
+ */
+let ContextDef;
+
+/**
+ * The external context given to React components to control whether they can
+ * render/play/etc.
+ *
+ * - renderable: whether this vDOM area is renderable. Analogous to
+ *   `display-locking` CSS.
+ * - playable: whether the playback is allowed in this vDOM area. If playback
+ *   is not allow, the component must immediately stop the playback.
+ *
+ * @return {!ContextDef}
+ */
+export function getAmpContext() {
+  if (self.__AMP_CONTEXT) {
```

Comment:
Why? If we are serializing/mapping them anyway, shouldn't we just use them as static constants?

Reply:
Hm, you're right. Because we're doing reserialization from a `JSON.stringify` from some parent node, we don't need these to be the same context instance across boundaries.